tests/collapse-dtrace.rs

```diff
@@ -0,0 +1,183 @@
+#[macro_use]
+extern crate pretty_assertions;
+
+extern crate inferno;
+
+use inferno::collapse::dtrace::{Folder, Options};
+use inferno::collapse::Collapse;
+use log::Level;
+use std::fs::{self, File};
+use std::io::{self, BufRead, BufReader, Cursor};
+use std::path::Path;
+
+fn test_collapse_perf(test_file: &str, expected_file: &str, options: Options) -> io::Result<()> {
+    let r = BufReader::new(
+        File::open(test_file).expect(&format!("Test file {} not found.", test_file)),
+    );
+    let expected_len = fs::metadata(expected_file)
+        .expect(&format!("Result file {} not found.", expected_file))
+        .len() as usize;
+    let mut result = Cursor::new(Vec::with_capacity(expected_len));
+    let mut perf = Folder::from(options);
+    let return_value = perf.collapse(r, &mut result);
+    let mut expected = BufReader::new(
+        File::open(expected_file).expect(&format!("Result file {} not found.", expected_file)),
+    );
+
+    result.set_position(0);
+
+    let mut buf = String::new();
+    let mut line_num = 1;
+    for line in result.lines() {
+        // Strip out " and ' since perl version does.
+        let line = line.unwrap().replace("\"", "").replace("'", "");
+
+        if expected.read_line(&mut buf).unwrap() == 0 {
+            println!("extra line: {}", line);
+            panic!(
+                "\noutput has more lines than expected result file: {}",
+                expected_file
+            );
+        }
+        assert_eq!(line, buf.trim_end(), "\n{}:{}", expected_file, line_num);
+        buf.clear();
+        line_num += 1;
+    }
+
+    if expected.read_line(&mut buf).unwrap() > 0 {
+        panic!(
+            "\n{} has more lines than output, beginning at line: {}",
+            expected_file, line_num
+        )
+    }
+
+    return_value
+}
```

Comment:
This function can probably be shared across _all_ our collapsers given that they all have a shared trait (`Collapse`). We should take advantage of that and factor this function out. /cc @jasonrhansen